IdentityCore/src/cache/accessor/MSIDDefaultTokenCacheAccessor.m

```diff
@@ -496,15 +497,24 @@ - (MSIDAccount *)getAccountForIdentifier:(MSIDAccountIdentifier *)accountIdentif
         if (wipeError) MSID_LOG_WITH_CTX_PII(MSIDLogLevelWarning, context, @"Failed to read wipe info with error %@", MSID_PII_LOG_MASKABLE(wipeError));
         return nil;
     }
+    
+    MSIDAccount *firstAccount = nil;
 
     for (MSIDAccountCacheItem *cacheItem in accountCacheItems)
     {
         MSIDAccount *account = [[MSIDAccount alloc] initWithAccountCacheItem:cacheItem];
-        if (account) return account;
+        if (!account) continue;
+
+        if (realmHint && [account.realm isEqualToString:realmHint])
+        {
+            return account;
```

Comment:
Can we add some comment here to say that we are trying to match the realm hint and in case we are not able to find any account that matches realm hint, we just return the first account in cache.
For instance, if we look at this code by point of view of trying to match the account with realm hint, it might seem wrong as in case of no match, we just return the first found account.

Reply:
Added :)